experimental/bridge/opentracing/lib.go

```diff
@@ -0,0 +1,558 @@
+	e.emitCommon(key, value)
+}
+
+func (e *bridgeFieldEncoder) EmitUint64(key string, value uint64) {
+	e.emitCommon(key, value)
+}
+
+func (e *bridgeFieldEncoder) EmitFloat32(key string, value float32) {
+	e.emitCommon(key, value)
+}
+
+func (e *bridgeFieldEncoder) EmitFloat64(key string, value float64) {
+	e.emitCommon(key, value)
+}
+
+func (e *bridgeFieldEncoder) EmitObject(key string, value interface{}) {
+	e.emitCommon(key, value)
+}
+
+func (e *bridgeFieldEncoder) EmitLazyLogger(value otlog.LazyLogger) {
+	value(e)
+}
+
+func (e *bridgeFieldEncoder) emitCommon(key string, value interface{}) {
+	e.pairs = append(e.pairs, otTagToOtelCoreKeyValue(key, value))
+}
+
+func otLogFieldsToOtelCoreKeyValues(fields []otlog.Field) []otelcore.KeyValue {
+	encoder := &bridgeFieldEncoder{}
+	for _, field := range fields {
+		field.Marshal(encoder)
+	}
+	return encoder.pairs
+}
+
+func (s *bridgeSpan) LogKV(alternatingKeyValues ...interface{}) {
+	fields, err := otlog.InterleavedKVToFields(alternatingKeyValues...)
+	if err != nil {
+		return
+	}
+	s.LogFields(fields...)
+}
+
+func (s *bridgeSpan) SetBaggageItem(restrictedKey, value string) ot.Span {
+	s.ctx.setBaggageItem(restrictedKey, value)
+	return s
+}
+
+func (s *bridgeSpan) BaggageItem(restrictedKey string) string {
+	return s.ctx.baggageItem(restrictedKey)
+}
+
+func (s *bridgeSpan) Tracer() ot.Tracer {
+	return s.tracer
+}
+
+func (s *bridgeSpan) LogEvent(event string) {
+	s.LogEventWithPayload(event, nil)
+}
+
+func (s *bridgeSpan) LogEventWithPayload(event string, payload interface{}) {
+	data := ot.LogData{
+		Event:   event,
+		Payload: payload,
+	}
+	s.Log(data)
+}
+
+func (s *bridgeSpan) Log(data ot.LogData) {
+	record := data.ToLogRecord()
+	s.LogFields(record.Fields...)
+}
+
+type BridgeTracer struct {
+	otelTracer oteltrace.Tracer
+
+	warnOnce sync.Once
+}
+
+var _ ot.Tracer = &BridgeTracer{}
+var _ ot.TracerContextWithSpanExtension = &BridgeTracer{}
+
+func NewTracer(tracer oteltrace.Tracer) *BridgeTracer {
+	return &BridgeTracer{
+		otelTracer: tracer,
+	}
+}
+
+func (t *BridgeTracer) StartSpan(operationName string, opts ...ot.StartSpanOption) ot.Span {
+	sso := ot.StartSpanOptions{}
+	for _, opt := range opts {
+		opt.Apply(&sso)
+	}
+	// TODO: handle links, needs SpanData to be in the API first?
+	bReference, _ := otSpanReferencesToBridgeReferenceAndLinks(sso.References)
+	// TODO: handle span kind, needs SpanData to be in the API first?
+	attributes, _, hadTrueErrorTag := otTagsToOtelAttributesKindAndError(sso.Tags)
+	checkCtx := migration.WithDeferredSetup(context.Background())
+	checkCtx2, otelSpan := t.otelTracer.Start(checkCtx, operationName, func(opts *oteltrace.SpanOptions) {
+		opts.Attributes = attributes
+		opts.StartTime = sso.StartTime
+		opts.Reference = bReference.ToOtelReference()
+		opts.RecordEvent = true
+	})
+	if checkCtx != checkCtx2 {
+		t.warnOnce.Do(func() {
+			// TODO: what to use for logging?
```

Comment:
I like how https://github.com/lightstep/lightstep-tracer-go/blob/master/event_handlers.go allows the application to set a handler for this. I guess it would be an SDK configuration item.

Reply:
For now I went with the handler installed through the `SetWarningHandler` function of the bridge tracer.